course/templatetags/base.py

```diff
@@ -31,23 +32,35 @@ def get_date(cont, key):
 def brand_name():
     return mark_safe(settings.BRAND_NAME)
 
+
 @register.simple_tag
 def brand_name_long():
     return mark_safe(settings.BRAND_NAME_LONG)
 
+
 @register.simple_tag
 def brand_institution_name():
     return mark_safe(settings_text('BRAND_INSTITUTION_NAME'))
 
+
+@register.simple_tag
+def course_alert(instance):
+    exercises = PendingSubmission.objects.get_exercise_names_if_grader_is_unstable(instance)
+    if exercises:
+        message = format_lazy(
+            _('GRADER_PROBLEMS_ALERT -- {exercises}'),
+            exercises=exercises,
+        )
+        return mark_safe('<div class="alert alert-danger">{}</div>'.format(message))
```

Comment:
I have a feeling that calling `.format()` in the end forces evaluation of the lazy string `message`. You should use `format_lazy()` here as well.

> A version of [str.format()](https://docs.python.org/3/library/stdtypes.html#str.format) for when **format_string, args, and/or kwargs** contain lazy objects. The first argument is the string to be formatted.

https://docs.djangoproject.com/en/4.2/ref/utils/#django.utils.text.format_lazy

`mark_save()` should not force evaluation of the lazy string. In the Django source code, its function definition includes the decorator `keep_lazy`. So using `mark_safe()` is OK.
https://docs.djangoproject.com/en/4.2/_modules/django/utils/safestring/#mark_safe